### PR TITLE
Added lab8 code

### PR DESCRIPTION
**Pull request by maxrt101**

### CODE REVIEW

File: src/main/java/com/maxrt/shoeshop/App.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        System.out.print("Unsorted: ");
+        manager.getShoes().forEach(s -> System.out.print(" " + s));
+
+        System.out.print("\nSorted by size: ");
+        manager.sortBySize(SortOrder.ASC)
+                .forEach(s -> System.out.print(" " + s));
+
+        System.out.print("\nSorted by type: ");
+        manager.sortByType(SortOrder.ASC)
+                .forEach(s -> System.out.print(" " + s));
```
Comment (zveres-ss) [inline src/main/java/com/maxrt/shoeshop/App.java]: never use single letter variable names, even in lambda's

File: src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+import com.maxrt.shoeshop.shoes.ShoeType;
+import com.maxrt.shoeshop.shoes.Shoes;
+
+public class ShoesManager {
+    private final List<Shoes> shoes = new ArrayList<>();
+
+    public ShoesManager() { }
+
+    public void addShoe(Shoes shoe) {
```
Comment (zveres-ss) [inline src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java]: Getters&Setters have to be replaced by Lombok annotation

File: src/main/java/com/maxrt/shoeshop/shoes/Shoes.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.maxrt.shoeshop.shoes;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor(access = AccessLevel.PUBLIC)
+public class Shoes {
+    protected ShoeType type;
```
Comment (zveres-ss) [inline src/main/java/com/maxrt/shoeshop/shoes/Shoes.java:12]: why these fields are protected?

File: src/main/java/com/maxrt/shoeshop/shoes/FlipFlops.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.maxrt.shoeshop.shoes;
+
+import lombok.ToString;
+
+@ToString(callSuper = true, includeFieldNames = true)
+public class FlipFlops extends Shoes {
+    public FlipFlops(final int size, final String manufacturer) {
+        super(ShoeType.SUMMER, size, manufacturer);
```
Comment (zveres-ss) [inline src/main/java/com/maxrt/shoeshop/shoes/FlipFlops.java:16]: this class has no extra fields comparing to the parent class - either add extra information to it or remove from the codebase

File: src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    public List<Shoes> sortByManufacturer(final SortOrder order) {
+        return sortByManufacturer(order, shoes);
+    }
+
+    public List<Shoes> sortByManufacturer(final SortOrder order, final List<Shoes> shoes) {
+        List<Shoes> sorted = new ArrayList<>(shoes);
+        sorted.sort((s1, s2) -> s1.getManufacturer().compareTo(s2.getManufacturer()));
+        if (order == SortOrder.DESC) {
+            Collections.reverse(sorted);
+        }
```
Comment (zveres-ss) [inline src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java]: awful approach from complexity view - the data is sorted twice in the worst case. Sorting of already sorted data is the hardest thing for almost all algorithms. Please fix

File: src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        return sortByType(order, shoes);
+    }
+
+    public List<Shoes> sortByType(final SortOrder order, final List<Shoes> shoes) {
+        List<Shoes> sorted = new ArrayList<>(shoes);
+        sorted.sort((s1, s2) -> s1.getType().compareTo(s2.getType()));
+        if (order == SortOrder.DESC) {
+            Collections.reverse(sorted);
+        }
+        return sorted;
```
Comment (zveres-ss) [inline src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java]: do not sort the same collection twice!

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            <artifactId>findbugs-maven-plugin</artifactId>
+            <version>3.0.0</version>
+            <type>maven-plugin</type>
+        </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>RELEASE</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (zveres-ss) [inline pom.xml]: what is the purpose of this dependency?

**Review by zveres-ss — CHANGES_REQUESTED** [review]

File: src/main/java/com/maxrt/shoeshop/shoes/Shoes.java (hunk 3 above)
Comment (maxrt101) [reply src/main/java/com/maxrt/shoeshop/shoes/Shoes.java:12]: Because those fields were protected in my UML diagram

File: pom.xml (hunk 7 above)
Comment (maxrt101) [reply pom.xml]: It was automatically added by the IDE, and I hadn't noticed that

File: src/main/java/com/maxrt/shoeshop/App.java (hunk 1 above)
Comment (maxrt101) [reply src/main/java/com/maxrt/shoeshop/App.java]: Noted and fixed

File: src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java (hunk 2 above)
Comment (maxrt101) [reply src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java]: Replaced `getShoes()` with `@Getter`

File: src/main/java/com/maxrt/shoeshop/shoes/FlipFlops.java (hunk 4 above)
Comment (maxrt101) [reply src/main/java/com/maxrt/shoeshop/shoes/FlipFlops.java:16]: Added extra fields to child classes

File: src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java (hunk 5 above)
Comment (maxrt101) [reply src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java]: Fixed

File: src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java (hunk 6 above)
Comment (maxrt101) [reply src/main/java/com/maxrt/shoeshop/manager/ShoesManager.java]: Fixed

**Pull request closed without merging**